File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum, curSum = float('-inf'), 0

        for i in range(len(nums)):
            curSum += nums[i]

            if curSum > maxSum:
                maxSum = max(maxSum, curSum)
            
            if curSum < 0:
                curSum = 0
        return maxSum

File: 0053-maximum-subarray/0053-maximum-subarray.java
import java.util.*;

class Solution {
    public int maxSubArray(int[] nums) {
        long maxSum = Long.MIN_VALUE;
        long curSum = 0;

        for(int i = 0; i< nums.length; i++){
            curSum += nums[i];

            if(curSum > maxSum){
                maxSum = curSum;
            }

            if (curSum < 0){
                curSum = 0;
            }
        }
        return (int)maxSum;
    }
}